common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java

```diff
@@ -104,15 +104,11 @@ Runnable takeTask() {
     }
 
     private void fetchFromScheduledTaskQueue() {
-        if (hasScheduledTasks()) {
-            long nanoTime = AbstractScheduledEventExecutor.nanoTime();
-            for (;;) {
-                Runnable scheduledTask = pollScheduledTask(nanoTime);
-                if (scheduledTask == null) {
-                    break;
-                }
-                taskQueue.add(scheduledTask);
-            }
+        long nanoTime = AbstractScheduledEventExecutor.nanoTime();
+        Runnable scheduledTask = pollScheduledTask(nanoTime);
+        while (scheduledTask != null) {
+            taskQueue.add(scheduledTask);
```

Comment:
is there a reason why this code is different from `SingleThreadEventExecutor.fetchFromScheduledTaskQueue`? Maybe we could move the common method to `AbstractScheduledEventExecutor` as a final package private method and introduce an abstract package private accessor for `taskQueue()`? @normanmaurer